examples/sample-input-app-gateway.tfvars

```diff
@@ -0,0 +1,120 @@
+# !NOTE! - These are only a subset of CONFIG-VARS.md provided as examples.
+# Customize this file to add any variables from 'CONFIG-VARS.md' whose default values you
+# want to change.
+
+# ****************  REQUIRED VARIABLES  ****************
+# These required variables' values MUST be provided by the User
+prefix   = "<prefix-value>" # this is a prefix that you assign for the resources to be created
+location = "<azure-location-value>" # e.g., "eastus2"
+# ****************  REQUIRED VARIABLES  ****************
+
+# !NOTE! - Without specifying your CIDR block access rules, ingress traffic
+#          to your cluster will be blocked by default. In a SCIM environment,
+#          the AzureActiveDirectory service tag must be granted access to port
+#          443/HTTPS for the ingress IP address. 
+
+# **************  RECOMMENDED  VARIABLES  ***************
+default_public_access_cidrs = [] # e.g., ["123.45.6.89/32"]
+ssh_public_key              = "~/.ssh/id_rsa.pub"
+# **************  RECOMMENDED  VARIABLES  ***************
+
+# Tags can be specified matching your tagging strategy.
+tags = {} # for example: { "owner|email" = "<you>@<domain>.<com>", "key1" = "value1", "key2" = "value2" }
+
+# Postgres config - By having this entry a database server is created. If you do not
+#                   need an external database server remove the 'postgres_servers'
+#                   block below.
+postgres_servers = {
+  default = {},
+}
+
+## Azure Application Gateway with Azure Web Application Firewall (WAF)
+create_app_gateway = true
+
+app_gateway_config = {
+  backend_host_name = "<your Application Gateway host name>"  ## leave empty to use Azure public DNS assigned host name
+  backend_trusted_root_certificate = [{
+    name = "<rootcert-name>"
+    data = "<path-to-rootcert>"  ## Required if key_vault_secret_id not set
+    key_vault_secret_id = "<key-vault-scret-id>" ## Required if data not set
+  }]
+  ssl_certificate = [{
+    name = "<listener-cert-name>"
+    data = "<path-to-listener-cert>"  ## Required if key_vault_secret_id not set
+    password = "<password>"  ## Required when data is set
+    key_vault_secret_id = "<key-vault-scret-id>" ## Required if data not set
+  }]
+  identity_ids              = ["<identity-id-with-access-to-key-vault>"] ## Required if key_vault_secret_id is set above. 
+  ## Example identity_ids value: `/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/example-resource-group/providers/Microsoft.ManagedIdentity/userAssignedIdentities/userAssignedIdentityValue`
+  backend_address_pool_fqdn = ["<your-ingress-nginx-loadBalancer-hostname>"]  ## Required to setup the backend pool. This list only accepts FQDN.
+}
+
+waf_policy = "<path-to-WAF-policy-json-file>"   ## Required to configure WAF with Application Gateway
```

Comment:
Should this not be just part of the app_gateway_config? If it's empty or null its ignored otherwise we include WAF

Reply:
Yes. I kept it separate as it gets created by a different resource but as of now we are only creating one WAF_policy per Application Gateway so this can included as part of  `app_gateway_config`. I'll update it.

Reply:
Updated in the latest commit change